mods/vessels/init.lua

```diff
@@ -29,10 +33,21 @@ local function get_vessels_shelf_formspec(inv)
 		if not invlist or invlist[i]:is_empty() then
 			formspec = formspec ..
 				"image[" .. vx .. "," .. vy .. ";1,1;vessels_shelf_slot.png]"
+		else
+			local stack = invlist[i]
+			if stack:is_empty() then
+			else
+				n_items = n_items + stack:get_count()
+			end
 		end
 		vx = vx + 1
 	end
-	return formspec
+	meta:set_string("formspec", formspec)
+	if n_items == 0 then
+		meta:set_string("infotext", S("Empty Vessel Shelf"))
+	else
+		meta:set_string("infotext", S("Vessel Shelf (" .. n_items .." items)"))
```

Comment:
`S("Vessel Shelf (@1 items)", n_items)`

Why, you might ask?
A: `S` is for translations, hence any value from 0 to ?? would have to be translated individually. `@1` inserts the number in the right place after translating.

Reply:
I see; done!